CMSgov/cms-carts-seds
# Update SASS and other UI related dependencies

### Description

Properly fixes #139816 

Updates SASS to 1.80.6 and migrates changes
Updates FontAwesome from 1.2 -> 6.6
Remove MaterialUI (Wasn't being used)
Updates date-fns from 3.6 -> 4.1
Updates jsonpath from 1.0.2 -> 1.1
Updates launchdarkley-sdk from 3.3.2 -> 3.6
Updates mathjs from 7.5 -> 13.2.1
Updates prop-types from 15.7.2 -> 15.8.1
Updates vite from 5.4.6 -> 5.4.11
Updates vite-tsconfig-paths from 4.3.2 -> 5.1.2 and migrates changes

---
### How to test

Run the app and take a peek! Icons should show properly, data should save and load, and calculations (Like those in Section 2) should calculate correctly

---
### Pre-review checklist
